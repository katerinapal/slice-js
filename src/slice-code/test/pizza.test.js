import {runAllCombosTests} from './helpers/utils'

// fit(
//   'tryCatch(false)',
//   require('./helpers/utils').snapSlice(require.resolve('./fixtures/try-catch'), ({tryCatch}) => {
//     return [tryCatch(false)]
//   })
// )

runAllCombosTests({
  filename: require.resolve('./fixtures/pizza'),
  methods: [
    {
      methodName: 'makePizza',
      possibleArguments: [
        [{type: 'cheese', size: 'Large'}],
      ],
    },
  ],
})
