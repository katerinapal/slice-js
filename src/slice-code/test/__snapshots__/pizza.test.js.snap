exports[`test makePizza({"type":"cheese","size":"Large"}) 1`] = `
"export { makePizza };

const getCrust = (size, crustType) => \`${ size }: ${ crustType || \'Hand tossed pizza\' }\`;

function makePizza(order) {
  const pizza = {
    type: order.type,
    crust: getCrust(order.size, order.crustType),
    crustEdge: order.crustEdge || \'Garlic Buttery Blend\',
    sauce: undefined,
    cheese: undefined,
    meats: undefined,
    veggies: undefined
  };

  return preparePizza(pizza);
}

function preparePizza(pizza) {
  return pizza;
}"
`;
